lua/lspconfig/util.lua

```diff
@@ -257,7 +257,13 @@ function M.server_per_root_dir_manager(make_config)
         return
       end
       local client = lsp.get_client_by_id(id)
-      if client and client.name == conf.name then
+      if
```

Comment:
I think the design about `clients` is not need ..there need a redesign and refactor. but it's in another pr.

Reply:
I supposed that maybe need to add test case coverage for this code to keep the behavior

Reply:
Test cases for client behavior do not exist in current lspconfig. Most of the testing on the client side is done in the core. Most of the tests of lspconfig are tests of some public APIs of util.

Reply:
Maybe copy some core test code. Then do some things on lspconfig. But there is a little trouble.

Reply:
yes, ic. I mean if we need to do some refactoring easily having test coverage will be better, in case the code break some logic.
but actually, I have write `lua` test before, so I don't know the effort

Reply:
>  if we need to do some refactoring easily having test coverage will be better, in case the code break some logic.

this is correct . but some history reason ? I don't know ,maybe one day in the future there will be a change here :)